ADAL/src/request/ADAcquireTokenSilentHandler.m

```diff
@@ -349,9 +366,21 @@ - (void)getAccessToken:(ADAuthenticationCallback)completionBlock
             return;
         }
     }
+    
+    BOOL enrollmentIdMatch = YES;
+    
+    // If token is scoped down to a particular enrollmentId and app is capable for True MAM CA, verify that enrollmentIds match
+    // EnrollmentID matching is done on the request layer to ensure that expired access tokens get removed even if valid enrollmentId is not presented
+    if ([self isCapableForMAMCA] && ![NSString msidIsStringNilOrBlank:item.enrollmentId])
+    {
+        enrollmentIdMatch = configuration.enrollmentId && [configuration.enrollmentId isEqualToString:item.enrollmentId];
```

Comment:
can it be simplified to be `enrollmentIdMatch = [item.enrollmentId isEqualToString:configuration.enrollmentId];`? as we already checked `item.enrollmentId` being non-nil :)

Reply:
sure, I opted for being more explicit to call out that if item.enrollmentId is not nil, and configuration.enrollmentId is nil, we should fail.

Reply:
I am ok with both :)